apps/passport-client/new-components/screens/Home/NewHomeScreen.tsx

```diff
@@ -0,0 +1,456 @@
+  gap: 12px;
+`;
+
+const TicketsContainer = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  gap: ${TICKET_VERTICAL_GAP}px;
+`;
+
+const positionInPx = (
+  currentPos: number,
+  elWidth: number,
+  len: number,
+  gap: number
+): number => {
+  const max = len * (elWidth + gap);
+  const truePos = currentPos * (elWidth + gap);
+
+  return truePos > max ? max : truePos;
+};
+
+const calculateElWidth = (
+  scrollWidth: number,
+  gap: number,
+  len: number
+): number => {
+  return Math.ceil((scrollWidth - gap * (len - 1)) / len);
+};
+
+const getEventDetails = (tickets: TicketPack): ITicketData => {
+  return tickets.eventTicket.claim.ticket as ITicketData;
+};
+
+const EmptyCardContainer = styled.div`
+  display: flex;
+  height: 302px;
+  justify-content: center;
+  align-items: center;
+  border-radius: 16px;
+  background: #e1e1e2;
+  /* shadow-inset-black */
+  box-shadow: 1px 1px 0px 0px rgba(0, 0, 0, 0.1) inset;
+  padding: 0 40px;
+`;
+
+const InnerContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  text-align: center;
+`;
+
+const calculateTicketsColumnHeight = (ticketRefs: HTMLDivElement[]): number => {
+  const len = ticketRefs.length;
+  // we have to calculate it per ticket column since different tickets can have different heights (if you have addons vs you dont)
+  const sumOfTicketsHeight = ticketRefs.reduce((acc, ref) => {
+    acc += ref.clientHeight;
+    return acc;
+  }, 0);
+
+  return sumOfTicketsHeight + len * TICKETS_HORIZONTAL_GAP;
+};
+
+const useWindowWidth = (): number => {
+  const [windoWidth, setWindowWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const onResize = (): void => {
+      setWindowWidth(window.innerWidth);
+    };
+
+    window.addEventListener("resize", onResize);
+
+    return () => {
+      window.removeEventListener("resize", onResize);
+    };
+  }, []);
+
+  return windoWidth;
+};
+
+const EmptyCard = (): ReactElement => {
+  return (
+    <EmptyCardContainer>
+      <InnerContainer>
+        <Typography fontWeight={800} color="var(--text-tertiary)">
+          YOU HAVE NO EVENT PASSES
+        </Typography>
+        <Typography color="var(--text-tertiary)">
+          Make sure you are logged in with the correct email address.
+        </Typography>
+      </InnerContainer>
+    </EmptyCardContainer>
+  );
+};
+
+export const NewHomeScreen = (): ReactElement => {
+  useSyncE2EEStorage();
+  const tickets = useTickets();
+  const [currentPos, setCurrentPos] = useState(0);
+  const [eventCardWidth, setEventCardWidth] = useState(0);
+  const [ticketCardWidth, setTicketCardWidth] = useState(0);
+  const [ticketsColumnHeight, setTicketsColumnHeight] = useState(0);
+  const dispatch = useDispatch();
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const pcdCardScrollRef = useRef<HTMLDivElement>(null);
+  const ticketsRef = useRef<Map<string, HTMLDivElement[]>>(new Map());
```

Comment:
this could be another potential source of the bug that prevents zupass from re-rendering when new tickets are added as part of the process that loads tickets from the server: https://www.notion.so/0xparc/ivan-s-feedback-on-PR-11271e0a5420806e97acc7fe5ed3ed60?pvs=4#11371e0a542080178a0dfd963fdf0221

Reply:
let's make sure to test this